ccdaservice/utils/date/date.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+function fDate(str, lim8 = false) {
+    const input = String(str);
+
+    if (lim8)
+        return input.substring(0, 8);
+
+    if (Number(input) === 0 || input.length === 1 || input === '0000-00-00')
+        return (new Date()).toISOString();
+
+    if (input.length === 8 || (input.length === 14 && Number(input.substring(12, 14)) === 0))
+        return `${input.slice(0, 4)}-${input.slice(4, 6)}-${input.slice(6, 8)}`;
+
+    // mm/dd/yyyy or mm-dd-yyyy
+    if (input.length === 10 && Number(input.substring(0, 2) <= 12))
+        return `${input.slice(6, 10)}-${input.slice(0, 2)}-${input.slice(3, 5)}`;
+
+    if (input.length === 17) {
+        const sections = input.split(' ');
+        return `${sections[0].slice(0, 4)}-${sections[0].slice(4, 6)}-${sections[0].slice(6, 8)} ${sections[1]}`;
+    }
+
+    if (input.length === 19 && (input[14] === '-')) {
+        const sections = input.split('-');
+        const date = `${sections[0].slice(0, 4)}-${sections[0].slice(4, 6)}-${sections[0].slice(6, 8)}`;
+        const time = `${sections[0].slice(8, 10)}:${sections[0].slice(10, 12)}:${sections[0].slice(12, 14)}`;
```

Comment:
The time code does not look right here. In original function it is slicing it from the original `str` (ie. `input` in this new function) instead of the `strZone` (ie. `sections` in the new function).

Reply:
The original code has an assumption that makes it work: the first occurrence of `-` is at `input[14]`. Therefore, taking the characters from the original string and from `sections[0]` produces the same result, as both refer to the same characters at the beginning of the string.

If this assumption didn't hold, then doing `str1 = strDate + '  ' + strTime + '-' + strZone[1];` in the original code wouldn't be correct, as `strZone[1]` would be repeating characters that were already included in the formatted string.

Reply:
thanks for the explanation. sounds good.